spec/std/process_spec.cr

```diff
@@ -169,4 +170,25 @@ describe Process do
     end
     buffer.to_s.lines.size.should eq(1000)
   end
+
+  it "executes the new process with exec" do
+    tmpfile = Tempfile.new("crystal-spec-exec")
+    tmpfile.close
+    tmpfile.unlink
+    File.exists?(tmpfile.path).should be_false
+
+    fork = Process.fork do
+      Process.exec("/usr/bin/touch", {tmpfile.path})
+    end
+    fork.wait
+
+    File.exists?(tmpfile.path).should be_true
+    tmpfile.unlink
+  end
+
+  it "checks for existence" do
+    Process.exists?(Process.ppid).should be_true
+    # We can't reliably check whether it ever returns false, since we can't predict
+    # how PIDs are used by the system.
```

Comment:
On Linux PID appear to increment and eventually overflow, which would avoid this issue of PID reuse, especially on a fast race? I don't know about other systems, but I hope they don't reuse a PID immediately after freeing...